.github/workflows/main.yml

```diff
@@ -0,0 +1,79 @@
+name: CI
+
+on: push
+
+jobs:
+  static:
+    name: Static Analysis
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-python@v1
+        with:
+          python-version: 3.8
+
+      - name: Install flake8
+        run: pip install flake8
+
+      - name: Run flake8
+        uses: suo/flake8-github-action@releases/v1
+        with:
+          checkName: 'static'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  testing:
+    name: Automated Testing
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+
+      - name: Cache pip dependencies
```

Comment:
why do we need to cache pip dependencies?

Reply:
For subsequent build performance

Reply:
This step complicates the understanding of how are we building the project while we don't have any reason to reduce build time. I'd rather not use it. Besides, it's cache - always tricky )

Reply:
Can you explain the potential complications? This is common practice both in terms of CI and in terms of GitHub Actions, both of which recommend caching dependencies, and it's pretty much automatically handled.

Reply:
The complication is: we have a piece of code we don't need. And while debugging deployment issues - it will be another step to read and check. 

At the same time, I have no strong preferences over this one. We can go with cache if you'd like